File: app/src/main/java/com/example/pam_project/features/categories/list/CategoryView.java
package com.example.pam_project.features.categories.list;

import java.util.List;

public interface CategoryView {

    void bindCategories(final List<CategoryInformation> model);

    void showAddCategory();

    void showCategoryForm(final long id);

    void showEmptyMessage();

    void onCategoriesSwap(final int draggedPosition, final int targetPosition);

    void onCategoriesReceivedError();
}

File: app/src/main/java/com/example/pam_project/features/categories/list/CategoryView.kt
package com.example.pam_project.features.categories.list

interface CategoryView {
    fun bindCategories(model: List<CategoryInformation?>?)
    fun showAddCategory()
    fun showCategoryForm(id: Long)
    fun showEmptyMessage()
    fun onCategoriesSwap(draggedPosition: Int, targetPosition: Int)
    fun onCategoriesReceivedError()
}